src/signup/router.py

```diff
@@ -1,83 +1,49 @@
 from pydantic import BaseModel
-from fastapi.responses import JSONResponse
-from src.core.dependencies import get_api_config
 
-config = get_api_config()
+from sqlalchemy import String, DateTime
+from sqlalchemy.orm import sessionmaker, Mapped, mapped_column
+
+from src.database import Base, connection
 
 router = APIRouter()
 
 
+
+class Attendees(Base):
+    __tablename__ = "attendees"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(String(50))
+    password: Mapped[str] = mapped_column(String(256))
+    email: Mapped[str] = mapped_column(String(50))
+    invoice: Mapped[str] = mapped_column(String(512))
+    created_at: Mapped[DateTime] = mapped_column(
+        DateTime,
+        default=datetime.utcnow
+    )
+
+
+
 class Signup(BaseModel):
     discord_id: str
     name: str
     password: str
     email: str
     invoice: str
-    created_at: datetime = datetime.now()
 
 
 @router.post("/signup", status_code=201)
-async def signup(signup: Signup):
-    request_body = {
-        "parent": {
-            "database_id": config.DATABASE_ID
-        },
-        "properties": {
-            "discord_id": {
-                "title": [
-                    {
-                        "text": {
-                            "content": signup.discord_id
-                        }
-                    }
-                ]
-            },
-            "name": {
-                "rich_text": [
-                    {
-                        "text": {
-                            "content": signup.name
-                        }
-                    }]
-            },
-            "password": {
-                "rich_text": [
-                    {
-                        "text": {
-                            "content": signup.password
-                        }
-                    }
-                ]
-            },
-            "email": {
-                "email": signup.email
-            },
-            "invoice": {
-                "rich_text": [
-                    {
-                        "text": {
-                            "content": signup.invoice
-                        }
-                    }
-                ]
-            }
-        }
-    }
-
-    status = httpx.post(
-        "https://api.notion.com/v1/pages",
-        headers={
-            "Content-Type": "application/json",
-            "Authorization": f"Bearer {config.AUTHORIZATION}",
-            "Notion-Version": "2022-06-28",
-        },
-        json=request_body
-    )
-    if status.status_code == 201:
-        return {"status": "ok"}
-
-    return JSONResponse(status_code=400, content={"status": "error"})
+async def signup(signup: Signup,
+                 session_connection: sessionmaker = Depends(connection)):
+    attendees = Attendees(
+        name=signup.name,
+        password=signup.password,
+              email=signup.email,
+              invoice=signup.invoice,)
+    session = session_connection()
```

Comment:
whenever a new request arrives, FastAPI will take care of:

Calling your dependency ("dependable") function with the correct parameters.
Get the result from your function.
Assign that result to the parameter in your path operation function.

I think you don't have to call it again 
https://fastapi.tiangolo.com/tutorial/dependencies/#declare-the-dependency-in-the-dependant